addons/nuxeo-platform-3d/nuxeo-platform-3d-core/src/main/java/org/nuxeo/ecm/platform/threed/service/ThreeDBatchUpdateWork.java

```diff
@@ -94,6 +93,29 @@ public boolean isIdempotent() {
         return false;
     }
 
+    @Override
+    public boolean isGroupJoin() {
+        // This is a GroupJoin work with a trigger that can be used on the last work execution
+        return true;
+    }
+
+    @Override
+    public String getPartitionKey() {
+        return computeIdPrefix(repositoryName, docId);
+    }
+
+    @Override
+    public void onGroupJoinCompletion() {
+        if ("DONE".equals(getStatus())) {
+            try {
+                openSystemSession();
+                fireThreeDConversionsDoneEvent(session.getDocument(new IdRef(docId)));
+            } finally {
+                cleanUp(true, null);
+            }
+        }
+    }
```

Comment:
Compared to the PictureViewsGenerationWork and VideoConversionWork where `onGroupJoinCompletion` simply fires the event, can yo explain why:
- you check the DONE status
- you need to open a system session instead of using the `session` from AbstractWork
- you need to clean up

Reply:
see my comment on master version [here](https://github.com/nuxeo/nuxeo/pull/3540/files#r346203183)

Reply:
The work is closing the session, so it needs to be reopened, I use the same pattern openSystemSession/cleanUp that is used many times in the class. There are multiple cases where the work exit before being in the "Done" state, to reproduce the same behavior we must be sure the work is in this state to fire the event.